src/main/java/io/deephaven/csv/parsers/DataType.java

```diff
@@ -0,0 +1,5 @@
+package io.deephaven.csv.parsers;
+
+public enum DataType {
+    BOOLEAN_AS_BYTE, BYTE, SHORT, INT, LONG, FLOAT, DOUBLE, DATETIME_AS_LONG, CHAR, STRING, TIMESTAMP_AS_LONG, CUSTOM
+}
```

Comment:
I'm wondering if an enum is not enough structure to handle the `CUSTOM` case. (The abstract visitor pattern may be useful for custom data types and return values). Not sure if we "care" right now, but maybe something we'll be interested in future if we think there is a strong story around custom types.

Reply:
Alternatively, if you know the column types of your data before-hand (and set explicit parsers), you don't care about DataType - so maybe it's a rare issue where CUSTOM isn't enough.

Reply:
Oh heck. Great point. Yeah, on one hand you only get a custom type if you set an explicit parser for it, so you already know what you're getting. On the other hand, it would be nice if the DataType you got back were extensible. I agree with you and I'm sort of torn myself. I was trying to make life a little easier for our users by giving them a data type they could switch on (which is easier for them than dispatching through a visitor).